src/cargo/util/rustc.rs

```diff
@@ -83,21 +88,46 @@ impl Rustc {
         }
     }
 
+    pub fn primary_unit_process_with(&self, path: impl AsRef<Path>) -> Option<ProcessBuilder> {
+        self.primary_unit_wrapper
+            .as_ref()
+            .filter(|wrapper| !wrapper.get_program().is_empty())
+            .map(|wrapper| {
+                let mut cmd = wrapper.clone();
+                // TODO: FIX HACK
```

Comment:
This is where having the primary_unit_rustc exist as a wrapper becomes a problem. `cargo fix` doesn't really want to use the `rustc_wrapper` either, it's only supposed to run fix for primary units so it makes sense to invoke it via the primary wrapper, but when you do that it turns it into a wrapper by adding the path (rustc) as an arg, but for fix we already added the path to the wrapper so you end up with `cargo clippy-driver rustc ...` at the end. This check prevents that extra incorrect insert from happening but we need to find a better way to do this.